src/lib.rs

```diff
@@ -1,251 +1,252 @@
-    fn build(&self, app: &mut App) {
-        app.init_resource::<PausedForBlockers>()
-            .add_event::<PickingEvent>()
-            .add_system_set_to_stage(
-                CoreStage::First,
-                SystemSet::new()
-                    .with_run_criteria(|state: Res<PickingPluginsState>| {
-                        simple_criteria(state.enable_interacting)
-                    })
-                    .with_system(
-                        pause_for_picking_blockers
-                            .label(PickingSystem::PauseForBlockers)
-                            .after(PickingSystem::UpdateIntersections),
-                    )
-                    .with_system(
-                        mesh_focus
-                            .label(PickingSystem::Focus)
-                            .after(PickingSystem::PauseForBlockers),
-                    )
-                    .with_system(
-                        mesh_selection
-                            .label(PickingSystem::Selection)
-                            .after(PickingSystem::Focus),
-                    )
-                    .with_system(
-                        mesh_events_system
-                            .label(PickingSystem::Events)
-                            .after(PickingSystem::Selection),
-                    ),
-            );
-    }
+/// Makes an entity pickable.
+#[derive(Bundle, Default)]
+pub struct PickableBundle {
+    /// The entity's configurable [`PickLayer`](focus::PickLayer)
+    pub pick_layer: focus::PickLayer,
+    /// Tracks entity [`Interaction`] state.
+    pub interaction: Interaction,
+    /// The entity's configurable [`FocusPolicy`]
+    pub focus_policy: FocusPolicy,
+    #[cfg(feature = "selection")]
+    /// Tracks entity [`PickSelection`](selection::PickSelection) state.
+    pub selection: selection::PickSelection,
+    #[cfg(feature = "highlight")]
+    /// Tracks entity [`PickHighlight`](highlight::PickHighlight) state.
+    pub highlight: highlight::PickHighlight,
 }
 
-pub struct HighlightablePickingPlugins;
-impl PluginGroup for HighlightablePickingPlugins {
-    fn build(&mut self, group: &mut PluginGroupBuilder) {
-        group.add(CustomHighlightPlugin(StandardMaterialHighlight));
-        group.add(CustomHighlightPlugin(ColorMaterialHighlight));
+/// Components needed to build a pointer
+#[derive(Bundle)]
+pub struct PointerBundle {
+    /// The pointer's unique [`PointerId`](pointer::PointerId).
+    pub id: pointer::PointerId,
+    /// Tracks the pointer's location.
+    pub location: pointer::PointerLocation,
+    /// Tracks the pointer's button press state.
+    pub click: pointer::PointerPress,
+    /// Tracks the pointer's interaction state.
+    pub interaction: output::PointerInteraction,
+    #[cfg(feature = "selection")]
+    /// Tracks whether the pointer's multiselect is active.
+    pub multi_select: selection::PointerMultiselect,
+}
+impl PointerBundle {
+    /// Create a new pointer with the provided `id`.
```

Comment:
It's not clear here how these should be generated, or how users can ensure they're unique.

Should you be using an atomic here instead of letting users pass in their own ID?

Reply:
`PointerID` is a type, see the documentation on the type. Users only need to use this if they don't use the provided defaults (mouse + touches), or they want to add a software pointer.